datastream/dataset.py

```diff
@@ -104,10 +105,18 @@ def __len__(self):
         return self.length
 
     def __str__(self):
-        return str('\n'.join(
-            [str(self[index]) for index in range(min(3, len(self)))]
-            + (['...'] if len(self) > 3 else [])
-        ))
+        sample_length = 3
+        sample = list(itemgetter(*range(min(sample_length, len(self)))))
+
+        content = '\n'.join(
+            list(map(
+                "    {}".format,
+                sample
+                + (['...'] if len(self) > len(sample) else [])
+            ))
+        )
+
+        return f"{self.__clas__.__name__}(\n{content}\n)"
```

Comment:
`self.__class__` :slightly_smiling_face:

Reply:
oops :P clearly I only sketched it up for the draft :)